examples/aliquot-sum.masm

```diff
@@ -0,0 +1,47 @@
+# x: input number
+# f: current factor candidate
+# a: aliquot sum
+
+begin                     # [x]
+  push.0 swap             # [x, a]
+  push.0                  # [f, x, a]
+  push.1
+  while.true
+    # increment f
+    add.1                 # [f, x, a]
+
+    # check whether f is a factor of x
+    dup.1 dup.1           # [f, x, f, x, a]
+    u32checked_mod        # [x % f, f, x, a]
+    push.0 u32checked_eq  # [x % f == 0, f, x, a]
```

Comment:
Do we need to use `u32checked_eq`? I think we could save a cycle and just use `eq`, since if the result of the mod is anything other than 0 we'll end up with a 0 anyway, so I think we don't really care about checking that it was a u32. Also, u32checked_mod should give us a u32 result, and 0 is a u32, so we should be safe anyway.